src/main/java9/org/sqlite/nativeimage/SqliteJdbcFeature.java

```diff
@@ -0,0 +1,149 @@
+package org.sqlite.nativeimage;
+
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeJNIAccess;
+import org.graalvm.nativeimage.hosted.RuntimeResourceAccess;
+import org.sqlite.*;
+import org.sqlite.core.DB;
+import org.sqlite.core.NativeDB;
+import org.sqlite.jdbc3.JDBC3DatabaseMetaData;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class SqliteJdbcFeature implements Feature {
+
+    @Override
+    public void beforeAnalysis(BeforeAnalysisAccess a) {
+        a.registerReachabilityHandler(
+                this::nativeDbReachable, method(SQLiteJDBCLoader.class, "initialize"));
+    }
+
+    private void nativeDbReachable(DuringAnalysisAccess a) {
+        registerResources(a);
+        registerJNICalls(a);
+    }
+
+    private void registerResources(DuringAnalysisAccess a) {
+        RuntimeResourceAccess.addResource(
+                SQLiteJDBCLoader.class.getModule(),
+                "META-INF/maven/org.xerial/sqlite-jdbc/pom.properties");
+        RuntimeResourceAccess.addResource(
+                SQLiteJDBCLoader.class.getModule(),
+                "META-INF/maven/org.xerial/sqlite-jdbc/VERSION");
+        RuntimeResourceAccess.addResource(
+                JDBC3DatabaseMetaData.class.getModule(), "sqlite-jdbc.properties");
+
+        // TODO need a smarter way to get this resource location
+        String libraryResource;
+        if (Platform.includedIn(Platform.WINDOWS_AMD64.class)) {
```

Comment:
That was indeed the idea, hence the TODO. I didn't have time to look at it too closely yet but it will probably need some refactoring still because there is currently no clear cut `String getLibraryResourcePathForCurrentOS()` method.

For android, I'm a bit confused how this works, specifically why this exists, and why android is treated specially:
https://github.com/xerial/sqlite-jdbc/blob/78633768a40c13d244799c2527f5cf20eb8ba3af/src/main/java/org/sqlite/core/NativeDB.java#L42-L52

Reply:
> That was indeed the idea, hence the TODO

Whoops, I missed that.

> For android, I'm a bit confused how this works, specifically why this exists, and why android is treated specially:

Seems like it was done over 10 years ago for a reason that seems no longer relevant:
https://github.com/xerial/sqlite-jdbc/commit/097530bd109143e1da48a901928aefda04fd5d10